receiver/jaegerreceiver/trace_receiver.go

```diff
@@ -189,51 +185,42 @@ func (jr *jReceiver) Start(_ context.Context, host component.Host) error {
 	jr.mu.Lock()
 	defer jr.mu.Unlock()
 
-	var err = componenterror.ErrAlreadyStarted
-	jr.startOnce.Do(func() {
-		if err = jr.startAgent(host); err != nil && err != componenterror.ErrAlreadyStarted {
-			return
-		}
+	if err := jr.startAgent(host); err != nil {
+		return err
+	}
 
-		if err = jr.startCollector(host); err != nil && err != componenterror.ErrAlreadyStarted {
-			return
-		}
+	if err := jr.startCollector(host); err != nil {
+		return err
+	}
 
-		err = nil
-	})
-	return err
+	return nil
 }
 
 func (jr *jReceiver) Shutdown(context.Context) error {
-	var err = componenterror.ErrAlreadyStopped
-	jr.stopOnce.Do(func() {
-		jr.mu.Lock()
-		defer jr.mu.Unlock()
-		var errs []error
-
-		if jr.agentServer != nil {
-			if aerr := jr.agentServer.Close(); aerr != nil {
-				errs = append(errs, aerr)
-			}
-		}
-		for _, processor := range jr.agentProcessors {
-			processor.Stop()
-		}
+	jr.mu.Lock()
```

Comment:
Don't think this locking is needed is it? Starting and Stopping components get run in same goroutine so I assume memory ordering should be ok. I think most components depend on this memory consistency as well.

Reply:
Separate issue/concern but good catch, should be removed

Reply:
Do you mind doing a PR

Reply:
#3041